File: 백준/Silver/1935. 후위 표기식2/후위 표기식2.cc
#include <iostream>
#include <stack>

using namespace std;

int main(void) {
    stack <double> s;
    int n;
    cin >> n;
    
    string equation;
    cin >> equation;
    
    int *operand = new int[n];
    for (int i = 0; i < n; i++) {
        cin >> operand[i];
    }
    
    for (auto x: equation) {
        double n1, n2;
        
        if (x >= 'A' && x <= 'Z') {
            s.push(operand[x - 'A']);
        } else {
            n2 = s.top();
            s.pop();
            n1 = s.top();
            s.pop();
            
            switch(x) {
                case '+':
                    s.push(n1 + n2);
                    break;
                case '-':
                    s.push(n1 - n2);
                    break;
                case '*':
                    s.push(n1 * n2);
                    break;
                case '/':
                    s.push(n1 / n2);
                    break;
            }
        }
    }
    cout << fixed;
    cout.precision(2);
    
    cout << s.top() << '\n';
    return 0;
}

File: 백준/Silver/1935. 후위 표기식2/후위 표기식2.py
n = int(input())
equation = input()
num_list = [0] * n

for i in range(n) :
    num_list[i] = int(input())
    
stack = []
for i in equation :
    if ('A' <= i <= 'Z') :
        stack.append(num_list[ord(i) - ord('A')])
    else :
        num2 = stack.pop()
        num1 = stack.pop()
        if (i == '+') :
            stack.append(num1 + num2)
        elif (i == '-') :
            stack.append(num1 - num2)
        elif (i == '*') :
            stack.append(num1 * num2)
        elif (i == '/') :
            stack.append(num1 / num2)
            
print("%.2f" %stack[0])